operatorcourier/api.py

```diff
@@ -136,3 +138,21 @@ def nest(source_dir, registry_dir):
 
     with TemporaryDirectory() as temp_dir:
         nest_bundles(yaml_files, registry_dir, temp_dir)
+
+
+def flatten(source_dir, dest_dir):
+    """
+    Given a directory containing different versions of operator bundles
+    (CRD, CSV, package) in separate version directories, this function
+    copies all files needed to the flattened directory
+
+    :param source_dir: the directory containing different versions
+    of operator bundles (CRD, CSV, package) in separate version directories
+    :param dest_dir: the flattened directory path where all necessary files are copied
+    """
+    os.makedirs(dest_dir, exist_ok=True)
+
+    file_paths_to_copy = get_flattened_files_info(source_dir)
+    if file_paths_to_copy:
+        for (src_file_path, new_file_name) in file_paths_to_copy:
+            copyfile(src_file_path, os.path.join(dest_dir, new_file_name))
```

Comment:
I think this logic should be inside the `flatten` module. Take a look at the patterns of the other API methods, they do not have any abstractions related to the specific use of the method itself. If they have additional code, it is strictly related to converting input to something that is digestible by the underlying module. Here it seems like this is actual implementation detail of flattening - creating a directory and copying files. I would prefer something simpler like:

1. `create output directory if not exists`
2. `flatten.flatten(input_dir, output_dir)`

The actual implementation of the flattening should be left up to the `flatten` module.